### ISSUE TITLE
Incorrect parsing of slug links inside Tip blocks

### ISSUE THREAD

**Issue by attilaantal** [telerik/docs-seed#14]
**Description**
Slug Links in Tip blocks are not interpreted/parsed correctly, hence the generated URL is invalid. Issue appears on multiple pages in the documentation.

**To Reproduce**
Visit any of the following pages, find the Tips that contain links and inspect the item.
[Batch Editing Overview](https://docs.telerik.com/devtools/aspnet-ajax/controls/grid/data-editing/edit-mode/batch-editing/overview)
[Batch Editing Client-Side API](https://docs.telerik.com/devtools/aspnet-ajax/controls/grid/data-editing/edit-mode/batch-editing/server-side-api)
[Data Editing - Custom Edit Forms](https://docs.telerik.com/devtools/aspnet-ajax/controls/grid/data-editing/edit-mode/custom-edit-forms)

**Additional context**
Probably the entire Documentation is affected, where slug links are included in Tip blocks.

**Comment by imtodor** [comment]
Works fine with docs.telerik.com/devtools/justmock/getting-started/quick-start (link to the article - https://github.com/telerik/justmock-docs/edit/master/getting-started/quick-start.md).

Can you check whether the slugs exist?

**Comment by marin-bratanov** [comment]
The JustMock article does not have the **custom alert blocks** from the ajax docs. My best guess is that the liquid tags within custom blocks are not evaluated at all. I think the fix we used to have for this was to make sure the custom blocks parsers either ran first, or ran last, perhaps Ianko Dzhemerenov will remember more, he wrote them.

The issue still manifests in the ajax docs, here is another article (see the `Which File Do I Need to Install` link in the second tip): [https://docs.telerik.com/devtools/aspnet-ajax/getting-started/first-steps#first-steps-with-ui-for-aspnet-ajax](https://docs.telerik.com/devtools/aspnet-ajax/getting-started/first-steps#first-steps-with-ui-for-aspnet-ajax).

Sample markup that causes the problem

```
>tip The following article can help you choose the installation type that is most suitable for your needs and preferences: [Which File Do I Need to Install]({%slug installation/which-file-do-i-need-to-install%}).
```

Resulting link `https://docs.telerik.com/devtools/aspnet-ajax/getting-started/%7B%slug%20installation/which-file-do-i-need-to-install%%7D`

Correct slug is present in the article: [https://github.com/telerik/ajax-docs/blob/master/installation/which-file-do-i-need-to-install.md](https://github.com/telerik/ajax-docs/blob/master/installation/which-file-do-i-need-to-install.md)

**Comment by imtodor** [comment]
Here is another example for working slug in a custom alert block -  https://docs.telerik.com/devtools/wpf/controls/radbulletgraph/radbulletgraph_overview

**Comment by marin-bratanov** [comment]
Here is another example of it not working in a custom alert block (see the second block): [http://kendobuild-staging.openstack.progress.com/blazor/components/window/size](http://kendobuild-staging.openstack.progress.com/blazor/components/window/size) 

There IS something wrong with this, even though it somehow works somewhere. The Blazor docs started out from the aspnet-core docs repo, not from the AJAX repo, so they can't have inherited something specific to a problem in ajax. Therefore, the fact that this works for WPF and JustMock is some fluke that I'd like to have fixed for the other repos.

**Comment by eyupyusein** [comment]
One more: https://docs.telerik.com/devtools/aspnet-ajax/controls/htmlchart/client-side-programming/events/onclientseriesclicked(deprecated).html

The  OnSeriesClick link in the red note does not work properly.

**Comment by pepinho24** [comment]
One more not parsed slug in https://docs.telerik.com/devtools/aspnet-ajax/controls/htmlchart/appearance-and-styling/legend-settings the Visual Templates link in the tip down in the Appearance section

**Comment by bsivanov** [comment]
We isolated that blocks except the default one (`>` or `>warning`) are breaking the slugs.

**Comment by dessyordanova** [comment]
Please refer to the following help article and have a look at the first 3 important fields: https://devdocs.telerik.com/devtools/winforms/controls/richtextbox/overview
It seems that a slug with a slash in it breaks the link.

**Comment by marin-bratanov** [comment]
Yes, it's the slash in a slug in a `>` block. I've started using dashes even though I have >100 articles with slashes already. Dashes are ok.

**Comment by dessyordanova** [comment]
Yes, the slashes are OK. But we have quite a huge number of articles with slashes in the slug.

**Comment by marin-bratanov** [comment]
oh, yes, in AJAX where I first hit this we have thousands (probably > 7000), in Blazor I have about 120, I think.

**Comment by imtodor** [comment]
Substituting slashes with dashes is easy, nevertheless the count of the slugs...Find/Replace with a regex will do the job.

**Comment by dessyordanova** [comment]
But this will require redirects which are not always so easy to implement as it sounds.

**Comment by imtodor** [comment]
> But this will require redirects which are not always so easy to implement as it sounds.

AFAIK, a slug is something like a 'key' for a given article? Why the redirect?

**Comment by dessyordanova** [comment]
I suppose because, on the website, the real URL (containing the slash) will be stored in the page, not the slug. If we change the slug, wouldn't it change the URL as well? Maybe I am wrong?

**Comment by pepinho24** [comment]
The issue seems to be related to the RegEx finding the slug inside the alert. It currently searches for "\w" and a dash:
- https://github.com/telerik/docs-seed/blob/master/_plugins/alert_generator.rb#L23

A potential fix is to enhance the regex to include some of the more common characters as well `/.*?(%7[Bb]%slug%20([0-9a-zA-Z_\-\(\)\*\.\/\,\%\'\?\:]+)%{2}7[Dd])/`

This covers all the following scenarios: https://rubular.com/r/5xkN2wCqUr4mWN
- "slug-with-dashes",
- "slug/with/forward/slashes",
- "slug.with.dots",
- "slug-with-(brackets)",
- "slug_with_underscore",
- "slug\*with\*stars",
- "slug1with2numbers3"
- "slug,with,comma"
- "slug%with%percentage"
- "slug:with:colon"
- "slug?with?question?mark"
- "slug'with'quote"

Here is a ruby file that can be used for testing
- [AlertGenerationRegexTests.rb.txt](https://github.com/telerik/docs-seed/files/7498938/AlertGenerationRegexTests.rb.txt)
